sktime/distances/mpdist.py

```diff
@@ -0,0 +1,275 @@
+import numpy as np
+import pandas as pd
+
+def sliding_dot_products(q, t, q_len, t_len):
+    """
+
+        Computes the sliding dot products between a query and a time series.
+
+        Parameters
+        ----------
+            q: pandas.Series
+                Query.
+            t: pandas.Series
+                Time series.
+            q_len: int
+                Length of the query.
+            t_len: int
+                Length of the time series.
+
+        Output
+        ------
+            dot_prod: numpy.array
+                        Sliding dot products between q and t.
+
+    """
+
+    # Reversing query and padding both query and time series
+    t_padded = np.pad(t, (0, t_len))
+    q_reversed = np.flipud(q)
+    q_reversed_padded = np.pad(q_reversed, (0, 2*t_len-q_len))
+
+    # Applying FFT to both query and time series
+    t_fft = np.fft.fft(t_padded)
+    q_fft = np.fft.fft(q_reversed_padded)
+
+    # Applying inverse FFT to obtain the convolution of the time series by the query
+    element_wise_mult = np.multiply(t_fft, q_fft)
+    reverse_fft = np.fft.ifft(element_wise_mult)
+
+    # Returns only the valid dot products from reverse_fft
+    dot_prod = reverse_fft[q_len-1:t_len].real
+
+    return dot_prod
+
+def compute_mean_std(q, t, q_len, t_len, n_t_subs):
+    """
+
+        Computes the mean and standard deviation of the query and the time series.
+
+        Parameters
+        ----------
+            q: pandas.Series
+                Query.
+            t: pandas.Series
+                Time series.
+            q_len: int
+                Length of the query.
+            t_len: int
+                Length of the time series.
+            n_t_subs: int
+                Number of subsequences in the time series.
+
+        Output
+        ------
+            q_mean: float
+                    Mean of the elements from the query.
+            q_std: float
+                    Standard deviation of the elements from the query.
+            t_mean: numpy.array
+                    Array with the mean of the elements from each subsequence of length(query) from the time series.
+            t_std: numpy.array
+                    Array with the standard deviation of the elements from each subsequence of length(query) from the time series.
+
+    """
+
+    # Query
+    q_mean = np.mean(q)
+    q_std = np.std(q)
+
+    # Time series
+    t_mean = np.empty(n_t_subs)
```

Comment:
numpy probably has a function to copy an array for you rather than doing it manually with a loop (plus numpy's version will be much quicker)